src/elements/ExactSbend.H

```diff
@@ -196,33 +201,35 @@ namespace impactx::elements
             } else {
 
                // assign intermediate quantities
-               amrex::ParticleReal const pperp2 = powi<2>(pt)-2.0_prt * m_ibeta * pt - powi<2>(py)+1.0_prt;
-               amrex::ParticleReal const px2 = powi<2>(px);
+               T_Real const pperp2 = powi<2>(pt)-2.0_prt * m_ibeta * pt - powi<2>(py)+1.0_prt;
+               T_Real const px2 = powi<2>(px);
                // determine if particle lies within the domain of map definition
-               if (pperp2 <= px2)
+               auto const mask = pperp2 <= px2;
+               if (stdx::any_of(mask) == true)
                {
-                   amrex::ParticleIDWrapper{idcpu}.make_invalid();
+                   amrex::ParticleIDWrapper<T_IdCpu>{idcpu}.make_invalid(mask);
                } else
                {
-                   amrex::ParticleReal const pperp = std::sqrt(pperp2);
-                   amrex::ParticleReal const pzi = std::sqrt(powi<2>(pperp) - powi<2>(px));
-                   amrex::ParticleReal const rho = m_rc + xout;
+                   T_Real const pperp = sqrt(pperp2);
+                   T_Real const pzi = sqrt(powi<2>(pperp) - powi<2>(px));
+                   T_Real const rho = m_rc + xout;
 
                    // update momenta
                    pxout = px * m_cos_phi + (pzi - rho / m_rc) * m_sin_phi;
                    pyout = py;
                    ptout = pt;
 
                    // angle of momentum rotation
-                   amrex::ParticleReal const px2f = powi<2>(pxout);
+                   T_Real const px2f = powi<2>(pxout);
                    // determine if particle lies within domain of map definition
-                   if (pperp2 <= px2f)
+                   auto const mask2 = pperp2 <= px2f;
+                   if (stdx::any_of(mask2) == true)
                    {
-                       amrex::ParticleIDWrapper{idcpu}.make_invalid();
+                       amrex::ParticleIDWrapper<T_IdCpu>{idcpu}.make_invalid(mask2);
                    } else
```

Comment:
Technically, we need to apply the else branch to all particles with `!mask2`...

Reply:
But probably ok to still to the `else` branch, as we mark the particles that might get Inf/NaN values as invalid anyway.